### PR TITLE
Get wider, detail information

### PR DESCRIPTION
**Pull request by nekop**
Initial improvement proposal.

- Get more data
- Better tmp dir handling
- Removed output flag for now as I'm not sure if having both json and yaml formats is good for troubleshooting

### CODE REVIEW

**Comment by nekop** [comment]
Example output:

```
$ oc plugin sos -n logging
Data capture complete and archived in /tmp/oc-sos-logging-20180206-210447.tar.xz
$ tar tf /tmp/oc-sos-logging-20180206-210447.tar.xz
logging/
logging/pods-logging-kibana-3-bwlw2_kibana-proxy.previous.log
logging/pods-logging-kibana-3-bwlw2_kibana-proxy.log
logging/pods-logging-kibana-3-bwlw2_kibana.previous.log
logging/pods-logging-kibana-3-bwlw2_kibana.log
logging/pods-logging-fluentd-rwgjv_fluentd-elasticsearch.previous.log
logging/pods-logging-fluentd-rwgjv_fluentd-elasticsearch.log
logging/pods-logging-fluentd-6fbm9_fluentd-elasticsearch.previous.log
logging/pods-logging-fluentd-6fbm9_fluentd-elasticsearch.log
logging/pods-logging-fluentd-4b5n5_fluentd-elasticsearch.previous.log
logging/pods-logging-fluentd-4b5n5_fluentd-elasticsearch.log
logging/pods-logging-fluentd-2fxhc_fluentd-elasticsearch.previous.log
logging/pods-logging-fluentd-2fxhc_fluentd-elasticsearch.log
logging/pods-logging-es-data-master-ye3u4nvd-3-q6md6_elasticsearch.previous.log
logging/pods-logging-es-data-master-ye3u4nvd-3-q6md6_elasticsearch.log
logging/pods-logging-es-data-master-ye3u4nvd-3-q6md6_proxy.previous.log
logging/pods-logging-es-data-master-ye3u4nvd-3-q6md6_proxy.log
logging/oc-get-all.txt
logging/oc-get-all.yaml
logging/oc-get-project.yaml
logging/oc-get-event.txt
logging/oc-status.txt
logging/oc-version.txt
```

**Comment by bostrt** [comment]
> Removed output flag for now as I'm not sure if having both json and yaml formats is good for troubleshooting

There are some stakeholders in this project that would like to see the JSON output. It would helpful for their parsing engine that may be feeding off of the data captured by this script.

Would you mind adding that back in? Submitting a follow-up commit in the same branch/PR would be fine.

**Comment by nekop** [comment]
Ok adding it back. I prefer yaml for the default output because yaml is more readable format to me. What do you think?

**Comment by nekop** [comment]
If the parsing tool doesn't understand the yaml format, what is the point we support the output flag with yaml?

**Comment by bostrt** [comment]
>  I prefer yaml for the default output because yaml is more readable format to me. What do you think?

Yes, I do typically prefer yaml when browsing configuration so would like to keep yaml support.

> If the parsing tool doesn't understand the yaml format, what is the point we support the output flag with yaml?

I think the flag should remain for user preference. The other stakeholders in this project will require the json output and others (like you and I) prefer the yaml output.

**Comment by nekop** [comment]
Thanks, I wish that kind of tools support yaml formats.

I'm adding secret/cm flags. Are you going to merge this one or can I add some more commits for the review at the same time?

**Comment by bostrt** [comment]
> I'm adding secret/cm flags. Are you going to merge this one or can I add some more commits for the review at the same time?

You can go ahead and keep committing to this PR. I'll review any new additions in the morning (EST).

File: sos.sh [hunk 1]
```diff
@@ -8,21 +8,46 @@ if [ $? -ne 0 ]; then
...
+# Enable command logging
+exec {BASH_XTRACEFD}>>$DEST/sos.log
+set -x
+
 # data capture
-oc get all,events -n $KUBECTL_PLUGINS_CURRENT_NAMESPACE -o "${KUBECTL_PLUGINS_LOCAL_FLAG_OUTPUT}" &> $DEST/kube-objects.${KUBECTL_PLUGINS_LOCAL_FLAG_OUTPUT}
-oc get all,events -n $KUBECTL_PLUGINS_CURRENT_NAMESPACE &> $DEST/kube-objects.txt
+oc version &> $DEST/oc-version.txt
+# Without -w, we cannot get full timestamps, so watch it and kill later
+oc get event -n $KUBECTL_PLUGINS_CURRENT_NAMESPACE -w &> $DEST/oc-get-event.txt &
```
Comment (bostrt) [inline sos.sh:22]: :+1:

**Review by bostrt — APPROVED** [review]

**Pull request merged**